### PR TITLE
Change leeways to TypeSignedDurationSecond

### PR DESCRIPTION
**Pull request by jasonodonnell**
Converting the custom leeway parameters to use the new `TypeSignedDurationSecond` type to support disabling leeway.

### CODE REVIEW

File: path_role.go [hunk 1]
```diff
@@ -74,21 +74,21 @@ duration specified by this value. At each renewal, the token's
 TTL will be set to the value of this parameter.`,
 			},
 			"expiration_leeway": {
-				Type: framework.TypeDurationSecond,
+				Type: framework.TypeSignedDurationSecond,
 				Description: `Duration in seconds of leeway when validating expiration of a token to account for clock skew. 
-Defaults to 150 (2.5 minutes), minimum of 1 second.`,
+Defaults to 150 (2.5 minutes) if set to 0 and can be disabled if set to -1.`,
```
Comment (mgaffney) [inline path_role.go]: Is there any value in adding tests for negative values to `path_role_test.go`?

Comment (jasonodonnell) [reply path_role.go]: Definitely valuable to have so I added it 👍
